src/modules/media/services/media.service.ts

```diff
@@ -414,6 +416,23 @@ export class MediaService {
 		});
 	}
 
+	public async countRelated(meemooIdentifiers: string[] = []): Promise<Record<string, number>> {
+		const items = await this.dataService.execute<GetObjectIdentifierTupleQuery>(
+			GetObjectIdentifierTupleDocument,
+			{
+				meemooIdentifiers,
+			}
+		);
+
+		const count: Record<string, number> = {};
+
+		items.data?.object_ie.forEach((item) => {
+			count[item.meemoo_identifier] = count[item.meemoo_identifier]++ || 1;
+		});
```

Comment:
Avoid the ++ operator outside of the for loop since its behavior can be confusing. In this case it has the wrong intended result. And all counts will be 1 instead of the desired count.
Since the client doesn't care, as long as the count is > 0 this isn't noticeable in the client.

Your line essentially equates to:
```
count[item.meemoo_identifier] = count[item.meemoo_identifier] = count[item.meemoo_identifier] + 1 || 1;
```

undefined + 1 => NaN => falsey

![image](https://user-images.githubusercontent.com/1710840/190356980-be4c9196-bbb6-4f34-9669-aafa0bd75406.png)

https://codepen.io/bertyhell/pen/zYjNrBw


```suggestion
		items.data?.object_ie.forEach((item) => {
			count[item.meemoo_identifier] = count[item.meemoo_identifier] ? count[item.meemoo_identifier] + 1 : 1;
		});
```